From c47a145396ac660e484032fb159197b82a320621 Mon Sep 17 00:00:00 2001
From: Martin Reboredo <yakoyoku@gmail.com>
Date: Fri, 21 May 2021 17:38:31 -0300
Subject: [PATCH] aom: Set fixed_qp_offsets to a deactivated value

aom only uses fixed_qp_offsets with the
Constant Quality (Q) Rate Control mode,
previously this was locking any usage
with another Rate Control mode.
---
 ext/aom/gstav1enc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ext/aom/gstav1enc.c b/ext/aom/gstav1enc.c
index 049460bf8..04e2e5ac1 100644
--- a/ext/aom/gstav1enc.c
+++ b/ext/aom/gstav1enc.c
@@ -426,6 +426,11 @@ gst_av1_enc_init (GstAV1Enc * av1enc)
   av1enc->tile_columns = DEFAULT_TILE_COLUMNS;
   av1enc->tile_rows = DEFAULT_TILE_ROWS;
 
+  av1enc->aom_cfg.fixed_qp_offsets[0] = -1;
+  av1enc->aom_cfg.fixed_qp_offsets[1] = -1;
+  av1enc->aom_cfg.fixed_qp_offsets[2] = -1;
+  av1enc->aom_cfg.fixed_qp_offsets[3] = -1;
+  av1enc->aom_cfg.fixed_qp_offsets[4] = -1;
   av1enc->aom_cfg.rc_dropframe_thresh = DEFAULT_DROP_FRAME;
   av1enc->aom_cfg.rc_resize_mode = DEFAULT_RESIZE_MODE;
   av1enc->aom_cfg.rc_resize_denominator = DEFAULT_RESIZE_DENOMINATOR;
-- 
2.30.2

